src/generator/AutoRest.Go/TransformerGo.cs

```diff
@@ -0,0 +1,258 @@
+                            mtm.PreparerNeeded = true;
+                        } else {
+                            mtm.PreparerNeeded = false;
+                        }
+                    }
+                });
+        }
+
+        private void TransformMethods(CodeModelGo cmg)
+        {
+            foreach (var mg in cmg.MethodGroups)
+            {
+                mg.Transform(cmg);
+            }
+
+            var wrapperTypes = new Dictionary<string, CompositeTypeGo>();
+            foreach (var method in cmg.Methods)
+            {
+                ((MethodGo)method).Transform(cmg);
+
+                var scope = new VariableScopeProvider();
+                foreach (var parameter in method.Parameters)
+                {
+                    parameter.Name = scope.GetVariableName(parameter.Name);
+                }
+
+                // fix up method return types
+                if (method.ReturnType.Body.ShouldBeSyntheticType())
+                {
+                    var ctg = new CompositeTypeGo(method.ReturnType.Body);
+                    if (wrapperTypes.ContainsKey(ctg.Name))
+                    {
+                        method.ReturnType = new Response(wrapperTypes[ctg.Name], method.ReturnType.Headers);
+                    }
+                    else
+                    {
+                        wrapperTypes.Add(ctg.Name, ctg);
+                        cmg.Add(ctg);
+                        method.ReturnType = new Response(ctg, method.ReturnType.Headers);
+                    }
+                }
+            }
+        }
+
+        private void FixStutteringTypeNames(CodeModelGo cmg)
+        {
+            // Trim the package name from exported types; append a suitable qualifier, if needed, to avoid conflicts.
+            var exportedTypes = new HashSet<object>();
+            exportedTypes.UnionWith(cmg.EnumTypes);
+            exportedTypes.UnionWith(cmg.Methods);
+            exportedTypes.UnionWith(cmg.ModelTypes);
+
+            var stutteringTypes = exportedTypes
+                                    .Where(exported =>
+                                        (exported is IModelType && (exported as IModelType).Name.FixedValue.StartsWith(cmg.Namespace, StringComparison.InvariantCultureIgnoreCase)) ||
+                                        (exported is Method && (exported as Method).Name.FixedValue.StartsWith(cmg.Namespace, StringComparison.InvariantCultureIgnoreCase)));
+
+            if (stutteringTypes.Count() > 0)
+            {
+                Logger.Instance.Log(Category.Warning, string.Format(CultureInfo.InvariantCulture, Resources.NamesStutter, stutteringTypes.Count()));
+                stutteringTypes.ForEach(exported =>
+                    {
+                        var name = exported is IModelType
+                                        ? (exported as IModelType).Name
+                                        : (exported as Method).Name;
```

Comment:
Do we have to do this because of shadowing? I don't run across examples of this too often, but dang, it makes me feel like Golang has the right approach to inheritance :p